base/setup/welcome/welcome.c

```diff
@@ -210,6 +210,101 @@ VOID TranslateEscapes(IN OUT LPTSTR lpString)
     }
 }
 
+/*
+ * Expands the path for the ReactOS Installer "reactos.exe".
+ * See also base/system/userinit/userinit.c!StartInstaller()
+ */
+VOID ExpandInstallerPath(IN OUT LPTSTR lpInstallerPath, IN SIZE_T PathSize)
+{
+    HANDLE hFind;
+    WIN32_FIND_DATA FindData;
+    SYSTEM_INFO SystemInfo;
+    PTSTR ptr;
+
+#if 0
+    if (_tcsicmp(lpInstallerPath, TEXT("reactos.exe")) != 0)
+        return;
+#endif
+
+    /*
+     * First, check whether we can find the installer from either the current
+     * ReactOS installation directory, or from our current directory.
+     */
+    *lpInstallerPath = UNICODE_NULL;
+    if (GetSystemWindowsDirectory(lpInstallerPath, PathSize - 12))
+        StringCchCat(lpInstallerPath, PathSize, L"\\");
+    StringCchCat(lpInstallerPath, PathSize, L"reactos.exe");
+
+    hFind = FindFirstFile(lpInstallerPath, &FindData);
```

Comment:
Why not use `GetFileAttributes` ?
Starting a search just to check if a file exists seems a bit overkill.

Reply:
https://blogs.msdn.microsoft.com/oldnewthing/20071023-00/?p=24713

OK, you would say, the file is not on a network drive anyway... (yet?! what about possible installation from a network?)

Reply:
That post argues _for_ GetFileAttributes, not against it.

Reply:
Yep, that's why I fixed my comment, just before you wrote yours .P